lib/SIL/Verifier/SILOwnershipVerifier.cpp

```diff
@@ -378,8 +395,13 @@ bool SILValueOwnershipChecker::gatherUsers(
                          << "Initial: " << *scopedOperand << "\n";
           });
         };
-        foundError |=
-            scopedOperand->getImplicitUses(nonLifetimeEndingUsers, &onError);
+
+        SmallVector<Operand *, 4> borrowLifetimeEndingUsers;
+        scopedOperand->getImplicitUses(borrowLifetimeEndingUsers, &onError);
+        nonLifetimeEndingUsers.append(borrowLifetimeEndingUsers.begin(),
+                                      borrowLifetimeEndingUsers.end());
+        reborrowVerifier.populateReborrowVerifier(value,
```

Comment:
Same here. I think we could just create a ReborrowVerifier here and just have it take the scopedOperand and process based off of that. Its a simpler thing to do than to try to process all of these at one time and will make it easier to debug.